worktree.go

```diff
@@ -286,6 +307,51 @@ func (w *Worktree) setHEADToBranch(branch plumbing.ReferenceName, commit plumbin
 	return w.r.Storer.SetReference(head)
 }
 
+// canSkipWorktreeUpdate determines if we can skip the worktree update
+// during checkout by comparing tree hashes. Returns true if current HEAD
+// and target commit point to the same tree, meaning no files need to change.
+func (w *Worktree) canSkipWorktreeUpdate(targetCommit plumbing.Hash) (bool, error) {
```

Comment:
```suggestion
func (w *Worktree) canSkipWorktreeUpdate(targetCommitHash plumbing.Hash) (bool, error) {
```

to be consistent with `indexMatchesTargetTree`